service/grails-app/services/org/olf/DependentModuleProxyService.groovy

```diff
@@ -0,0 +1,88 @@
+package org.olf
+
+import com.k_int.okapi.OkapiClient
+import groovy.util.logging.Slf4j
+import groovyx.net.http.HttpException
+import org.olf.general.Org
+
+
+/**
+ * This class acts as a facade for any optional modules that this module would depend on. The initial use case
+ * is for mod-vendors. An install might not have mod-vendors installed, in which case this module will only use
+ * it's own orgs store. If mod-vendors is installed however, it will be consulted prior to creating a new vendor
+ * record. If a record is found, a local proxy will be created which holds a reference to the upstream vendor record.
+ * This pattern allows us to integrate with optional third party modules where they are present, but continue
+ * if they are not. It also gives us a hook onto which we can cache information for performance reasons.
+ */
+@Slf4j
+public class DependentServiceProxyService {
```

Comment:
Is this class name different to the filename (DependentModuleProxyService.groovy) Is it possible to have them harmonized?

Reply:
Eurgh - it's the same in mod-agreements - I'll change it there!